Installation verification steps (#223)
blockfrost/blockfrost-platform: docs/src/content/installation/verification.md

# Verifying that you are up and running

## Prerequisites verification

Before verifying your Blockfrost platform operation, make sure that these critical configurations are correct:

### NFT address configuration

**Important:** You must use the exact address containing your Icebreaker NFT, not your general wallet address.

When running the init command or configuring manually, use the specific address containing the NFT:

```shell
--reward-address addr1xxx...  # Address containing NFT, not general wallet address
```

Your wallet likely contains multiple addresses.
The platform requires specifically the address where the NFT resides.
Check your wallet to identify which address holds the Icebreaker NFT.

> ### Network Port Configuration

Reviewer: ```suggestion
### Network port configuration
```

Make sure that your required ports are properly opened.
The default Blockfrost platform port is `3000`.

#### For cloud hosting

- Azure: Create a new outbound rule allowing port 3000 traffic

- AWS: Configure security groups to permit port 3000 traffic

- Other providers: Update firewall settings accordingly.

## Basic service verification

After confirming prerequisites, verify the service itself:

### Installation verification

Confirm that the software is properly installed:

```shell
blockfrost-platform --version
```

This should return the current version number.

### Service startup verification

Start the service and check for successful initialization:

```shell
blockfrost-platform --network mainnet \
                    --node-address 127.0.0.1 \
                    --secret your_icebreaker_secret \
                    --reward-address your_nft_address
```

In the logs, you should see initialization messages.
The message `DEBUG: Decoding done` indicates normal operation.

If you don't see this message or encounter errors, check:

- Node socket path accessibility
- Network configuration

> - Secret key validity

Reviewer: ```suggestion
- Secret key validity.
```

> ### Operational Verification

Reviewer: ```suggestion
### Operational verification
```

Once the service is running, verify that it is functioning correctly:

### Understanding the UUID

The Blockfrost platform generates a new UUID each time it starts.
This UUID is critical for accessing endpoints:

### Example log entry showing UUID
INFO: Your instance ID: 3fa85f64-5717-4562-b3fc-2c963f66afa6

Make a note of this UUID as you'll need it for subsequent verification steps.
Remember that this changes every time you restart the service.

> ### Metrics Endpoint Verification

Reviewer: ```suggestion
### Metrics endpoint verification
```

Check if your service is properly reporting metrics:

```shell
wget http://your-server-ip:3000/3fa85f64-5717-4562-b3fc-2c963f66afa6/metrics
```

Replace the UUID in the URL with your current instance ID.
If functioning properly, this endpoint provides:

- API calls served count
- Resource utilization statistics
- Uptime information.

Note: The metrics endpoint may require the latest binary.
If you receive a 500 error, update your installation.

> ## Grafana Dashboard Monitoring

Reviewer: ```suggestion
## Grafana dashboard monitoring
```

For a visual verification of your node's status, access the Blockfrost Grafana dashboard:

```shell
https://blockfrost.grafana.net/public-dashboards/8d618eda298d472a996ca3473ab36177
```

> Your node will appear as "Icebreaker X" where X is your assigned number (visible in your NFT metadata).

Reviewer: ```suggestion
Your node will appear as 'Icebreaker X' where X is your assigned number (visible in your NFT metadata).
```

This dashboard provides:

- Connection status
- Performance metrics
- Comparison with other Icebreaker nodes.

> ## Troubleshooting Common Issues

Reviewer: ```suggestion
## Troubleshooting common issues
```

If verification fails, check these common issues:

> ### Not seeing "Decoding done" message

Reviewer: ```suggestion
### Not seeing the 'Decoding done' message
```

# Verifying that you are up and running

## Prerequisites verification

> 

Reviewer 2: Maybe we should add running the cardano node is a prerequisite? I think it asks for the socket location.
@joseph-fajen
